app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
```

Comment:
```suggestion
class UsersController < ApplicationController
  before_action :find_user, only: %i(edit update destroy)
```

ActiveRecord find throws an error if a non-existent id is passed as the param, prefer to handle this and boot them. Additionally to prevent record enumeration, rather respond with unauthorized than bad_request.

> def find_user
>   @user ||= User.find(params[:id])
> rescue ActiveRecord::NotFoundError
>   head :unauthorized
> end